main.tf

```diff
@@ -6,7 +6,7 @@ locals {
 
   # s3_* - to get package from S3
   s3_bucket         = var.s3_existing_package != null ? lookup(var.s3_existing_package, "bucket", null) : (var.store_on_s3 ? var.s3_bucket : null)
-  s3_key            = var.s3_existing_package != null ? lookup(var.s3_existing_package, "key", null) : (var.store_on_s3 ? element(concat(data.external.archive_prepare.*.result.filename, [null]), 0) : null)
+  s3_key            = var.s3_existing_package != null ? lookup(var.s3_existing_package, "key", null) : (var.store_on_s3 ? format("%s/%s", dirname(element(concat(data.external.archive_prepare.*.result.filename, [null]), 0)), basename(element(concat(data.external.archive_prepare.*.result.filename, [null]), 0))) : null)
```

Comment:
On second thought, using `dirname()` and `basename()` is probably going to cause issues on Windows...

Reply:
Yes, it probably will not work there. I am going to close this PR.

Reply:
@antonbabenko I changed the method to strip the leading and opened a new PR: https://github.com/terraform-aws-modules/terraform-aws-lambda/pull/191